README.md

```diff
@@ -1,8 +1,35 @@
-This plugin does a HTTP/HTTPS request to a speficied URL with the build parameters automatically
-part of the payload as key/value pairs.
+# Http Request Plugin for Jenkins
 
-It can be configured whether GET or POST (default) should be used as HTTP method, as well
-as if the status return code of the response marks the build as failed (default) or is ignored.
-You can specify different username / password pairs in the global configuration to support
-authenticated HTTP/HTTPS requests, which you then refer to from your build configuration.
+This plugin sends a HTTP/HTTPS request to a speficied URL, with the build parameters optionally
+appended to the URL as key value pairs.
+
+## Features:
+
+* Select the HTTP method: GET, POST, PUT, DELETE, or HEAD
+* Select a range of expected response codes (a response code outside the range fails the build)
+* Specify a username / password in the global configuration, that can be used for Basic Digest Authentication
+* Supports From Authentication
+* Specify a string that must be present in the response, if the string is not present, the build fails
+* Store the response to a file
+* Set a connection timeout limit (build fails if timeout is exceeded)
+* Send an "Accept" header
+* Send a "Content-type" header
+
+## When using this plugin as a Pipeline step:
+
+* The build parameters configuration is ignored, you must append build parameters programmatically to the url. This is more flexible as you can cherry pick which build parameters to pass.
+
+### Pipeline example:
+
+In a Pipeline job, you have total control over how the url is
+formed. Suppose you have a build parameter called "pretty",
+you can pass it to the HTTP request programmatically with:
+
+    node () {
+        step([$class:
```

Comment:
@martinda 

> Do I need to have a dedicated step before I can write a pipeline step?

No. You don't need a dedicated step to write a test. Just use the general `step`.